pyuvdata/uvbase.py

```diff
@@ -356,8 +357,12 @@ def check(
                                         " be: " + str(param.expected_type)
                                     )
                         else:
-                            # Array
-                            if not isinstance(param.value.item(0), param.expected_type):
+                            # Array or quantity
+                            if isinstance(param.value, Quantity):
+                                check_val = param.value.item(0).value
+                            else:
+                                check_val = param.value.item(0)
```

Comment:
I think this will pass if the parameter is set to be an Astropy Quantity but the expected type is "float". Do we want that? Or should we make it so that the expected type should be a Quantity to pass? I'm concerned that in pyradiosky we actually set the expected type to be a Quantity in some cases and that might fail here?

Reply:
oh yeah that is an interesting point. This is more about evaluating the type of quantity, and just letting quantities slide through unnoticed but still supported. :thinking: I'm more than willing to remove it.

Reply:
how do you feel about this. I'm not sure the logic is 100% there at the moment, but basically it adds an additional check if the `expected_type` is a subclass of Quantity, if it is not and the `expected_type` is a data type like `float`, `int`, etc, then it will fall back to the other check.


<details>

```
                            if isinstance(param.value, Quantity):
                                # user put expected type as a type of quantity object
                                # not a more generic type of number.
                                if issubclass(param.expected_type, Quantity):
                                    # Verify the param is an instance
                                    # of the specific Quantity type
                                    if not isinstance(param.value, param.expected_type):
                                        raise ValueError(
                                            f"UVParameter {param} is a Quantity object "
                                            "but not the appropriate type. "
                                            f"Is {type(param.value)} but "
                                            f"expected {param.expected_type}."
                                        )
                                   else:
                                       continue
                                else:
                                    # expected type is not a type of Quantity
                                    # assuming it is a date a type, continue with check below
                                    check_val = param.value.item(0).value
                            else:
                                check_val = param.value.item(0)
                            if not isinstance(check_val, param.expected_type):
                                raise ValueError(
                                    "UVParameter " + p + " is not the appropriate"
                                    " type. Is: "
                                    + str(param.value.dtype)
                                    + ". Should be: "
                                    + str(param.expected_type)
                                )
```
</details>

Reply:
That looks reasonably good to me. I'll note that we should use `param.name` or `p` to in the print statement rather than just `param` because that gives a useless nonsense string (I actually changed that in #967 because I was trying to debug and couldn't figure out which parameter it was erroring on).

I'm a little confused about the part that reads:
```
# expected type is not a type of Quantity
# assuming it is a date a type, continue with check below
check_val = param.value.item(0).value
```

I think there's a typo in the first comment line, but I think you're assuming it's a Time object? I don't really understand why...

Reply:
I was thinking more like you have an array of quantities, but you set the expected type to `float` not `Quantity`. That would trigger the case with the comment you just mentioned. Maybe the overloading of the word "type" isn't helping here.

Reply:
ok, so it has nothing to do with a Time object? I guess I was confused by the comment line that read "assuming it is a date a type"

Reply:
correct, I was trying to find words to describe the different kids of numbers (float,int, etc), data types is what came to mind.